core/src/main/scala/filodb.core/memstore/PartitionSet.scala

```diff
@@ -561,6 +563,7 @@ final class PartitionSet(as: Array[FiloPartition], bs: Array[Byte], n: Int, u: I
     } else {
       cfor(0)(_ < buckets.length, _ + 1) { i =>
         if (buckets(i) == 3 && rhs(items(i))) {
+          items(i) = null // allow item to be garbage collected
```

Comment:
Yes, this allows the memory to be freed sooner. Leaving the object reference in the set means that it might live forever. There's no guarantee that this slot will ever be re-used.